packages/core-data/src/test/selectors.js

```diff
@@ -275,3 +276,45 @@ describe( 'getCurrentUser', () => {
 		expect( getCurrentUser( state ) ).toEqual( currentUser );
 	} );
 } );
+
+describe( 'getReferenceByDistinctEdits', () => {
+	it( 'should return referentially equal values across empty states', () => {
+		const state = { undo: [] };
+		expect( getReferenceByDistinctEdits( state ) ).toBe( getReferenceByDistinctEdits( state ) );
+
+		const beforeState = { undo: [] };
+		const afterState = { undo: [] };
+		expect( getReferenceByDistinctEdits( beforeState ) ).toBe( getReferenceByDistinctEdits( afterState ) );
+	} );
+
+	it( 'should return referentially equal values across unchanging non-empty state', () => {
+		const undoStates = [ {} ];
+		const state = { undo: undoStates };
+		expect( getReferenceByDistinctEdits( state ) ).toBe( getReferenceByDistinctEdits( state ) );
+
+		const beforeState = { undo: undoStates };
+		const afterState = { undo: undoStates };
+		expect( getReferenceByDistinctEdits( beforeState ) ).toBe( getReferenceByDistinctEdits( afterState ) );
+	} );
+
+	describe( 'when adding edits', () => {
+		it( 'should return referentially different values across changing states', () => {
+			const beforeState = { undo: [ {} ] };
+			beforeState.undo.offset = 0;
```

Comment:
Can you elaborate what's happening here? I can't figure out myself which I see as a good reason to fix with an inline comment.

Reply:
The `undo` reducer manually manages this `offset` property:

https://github.com/WordPress/gutenberg/blob/52ca63ee461848a742addac3e78a9fc0f637b5b5/packages/core-data/src/reducer.js#L304-L313

which is then used in selectors:

https://github.com/WordPress/gutenberg/blob/52ca63ee461848a742addac3e78a9fc0f637b5b5/packages/core-data/src/selectors.js#L293-L307

This particular case, as is in its name, mimics the addition of an edit state. So, aside from expanding the `undo` array, we also bump `offset` to point to the latest state.

The second test mimics the undo by just lowering the offset. Although I now see that there should be two objects in `undo` instead of one, for fidelity. :)

Reply:
It sounds like a very fragile construct but I guess this is what it is. I hope all those tests where undo state is manually tweaked won't produce too many issues when we try to change its internal shape in the future. 

I assume it's good as is :)